test/Reader/ReaderTest.php

```diff
@@ -309,6 +319,37 @@ public function testImportStringMethodThrowProperExceptionOnEmptyString()
         $feed = Reader\Reader::importString($string);
     }
 
+    public function testSetHttpFeedClient()
```

Comment:
Opted to keep them separate, as they test separate behavior.

`testSetHttpFeedClient()` is testing that injecting a `Zend\Feed\Reader\Http\ClientInterface` instance returns it verbatim. `testSetHttpClientWillDecorateAZendHttpClientInstance()` is testing that passing a `Zend\Http\Client` will decorate it in a `Zend\Feed\Reader\Http\ZendHttpClientDecorator` instance.